### PR TITLE
Multi query catchable

### PR DESCRIPTION
**Pull request by spillevink**
See #54

### CODE REVIEW

File: src/query.coffee [hunk 1]
```diff
@@ -32,7 +32,12 @@ class Query extends EventEmitter
       @emit 'error', err
 
   onRowDescription: (msg) ->
-    throw new errors.VerticaError("Cannot handle multi-queries with a callback!") if @callback? && @status?
+    if @status
+      err = new errors.VerticaError("Cannot handle multi-queries with a callback!")
+      if @callback
+        @error = err
+      else
+        @emit 'error', err
```
Comment (wvanbergen) [inline src/query.coffee]: I think it currently works fine if you're not using the callback method, but use an event listener instead. So we probably don't want to emit an error in this case.

Comment (spillevink) [inline src/query.coffee]: @wvanbergen: Thank you, PR updated!

File: test/functional/query_test.coffee [hunk 2]
```diff
@@ -101,6 +101,13 @@ describe 'Vertica.Connection#query', ->
       assert.ok typeof err, 'string'
       done()
 
+    it "should return multi-queries error", (done) ->
+      sql = "SELECT 1; SELECT 1"
+      connection.query sql, (err, _) ->
+        assert.ok typeof err, 'string'
+        assert.equal err.message, "Cannot handle multi-queries with a callback!"
+      done()
```
Comment (wvanbergen) [inline test/functional/query_test.coffee]: `done()` should be indented.

Comment (wvanbergen) [inline test/functional/query_test.coffee]: Also, I think the connection will be in a broken state after this. Is that acceptable?

 ``` coffeescript
connection.query "SELECT 1; SELECT 1", (err, _) ->
  assert.ok typeof err, 'string'
  assert.equal err.message, "Cannot handle multi-queries with a callback!"
  connection.query "SELECT 1", (err, result) ->
    // should this work?
```

File: test/functional/query_test.coffee [hunk 3]
```diff
@@ -101,6 +101,13 @@ describe 'Vertica.Connection#query', ->
       assert.ok typeof err, 'string'
       done()
 
+    it "should return multi-queries error", (done) ->
+      sql = "SELECT 1; SELECT 1"
+      connection.query sql, (err, _) ->
+        assert.ok typeof err, 'string'
+        assert.equal err.message, "Cannot handle multi-queries with a callback!"
+        connection.query "SELECT 1", (err, _) -> done err
```
Comment (spillevink) [inline test/functional/query_test.coffee]: @wvanbergen: This works, so it does not seem to be broken/stale.

Comment (wvanbergen) [inline test/functional/query_test.coffee]: :+1:

**Pull request merged**